dlp/src/test/java/dlp/snippets/JobsTests.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.concurrent.ExecutionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(JUnit4.class)
+public class JobsTests {
+
+    private static final Pattern JOB_ID_PATTERN = Pattern.compile("projects/.*/dlpJobs/i-\\d+");
+    private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+    private static final String GCS_PATH = System.getenv("GCS_PATH");
+    private static final String PUB_SUB_TOPIC_ID = "dlp-tests";
+    private static final String PUB_SUB_SUBSCRIPTION_ID = "dlp-test";
+    private ByteArrayOutputStream bout;
+
+    private static void requireEnvVar(String varName) {
+        assertNotNull(
+                String.format("Environment variable '%s' must be set to perform these tests.", varName),
+                System.getenv(varName));
+    }
+
+    @Before
+    public void checkRequirements() {
+        requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+        requireEnvVar("GOOGLE_CLOUD_PROJECT");
+        requireEnvVar("GCS_PATH");
+    }
+
+    @Before
+    public void setUp() {
+        bout = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(bout));
+
+    }
+
+    
+    @After
+    public void tearDown() {
+        System.setOut(null);
+        bout.reset();
+    }
+
+    @Test
+    public void testListJobs() throws Exception {
+        // Ensure that there is at least one job to list
+        InspectGcsFile.inspectGcsFile(PROJECT_ID,
```

Comment:
If there isn't a job, we should make sure there is with a `setUp` function that creates one that runs first.

Or, alternatively make the test pass if no jobs are found.

Reply:
I think the `setUp` approach is better, since the delete method is tested in the same class, and that has to have an existing job so that we can test it.